Gillian-C/lib/gilgen.ml

```diff
@@ -130,18 +130,112 @@ let trans_binop_expr ~fname binop te1 te2 =
   | Omulfs -> call BinOp_Functions.mulfs
   | _ -> failwith "Unhandled"
 
-let rec trans_expr ~fname ~local_env expr =
-  let trans_expr = trans_expr ~fname ~local_env in
+let rec num_to_integer_h n index =
+  let open BinNums in
+  match n with
+  | Coq_xI x ->
+      int_of_float (2. ** float_of_int index) + num_to_integer_h x (index + 1)
+  | Coq_xO x -> num_to_integer_h x (index + 1)
+  | Coq_xH -> int_of_float (2. ** float_of_int index)
+
+let num_to_integer n = num_to_integer_h n 0
+
+(* let pp_coq_type fmt typ =
+   let open Ctypes in
+   match typ with
+   | Tvoid -> Fmt.pf fmt "Tvoid"
+   | Tint _ -> Fmt.pf fmt "Tint"
+   | Tlong _ -> Fmt.pf fmt "Tlong"
+   | Tfloat _ -> Fmt.pf fmt "Tfloat"
+   | Tpointer _ -> Fmt.pf fmt "Tpointer"
+   | Tarray _ -> Fmt.pf fmt "Tarray"
+   | Tfunction _ -> Fmt.pf fmt "Tfunction"
+   | Tstruct _ -> Fmt.pf fmt "Tstruct"
+   | Tunion _ -> Fmt.pf fmt "Tunion" *)
+
+let rec pp_expr fmt expr =
+  match expr with
+  | Evar id -> Fmt.pf fmt "Evar id NUM[%d]" (num_to_integer id)
+  | Eaddrof _id -> Fmt.pf fmt "Eaddrof id"
+  | Econst _constant -> Fmt.pf fmt "Econst constant"
+  | Eunop (_unop, e) -> Fmt.pf fmt "Eunop unop (%a)" pp_expr e
+  | Ebinop (_binop, e1, e2) ->
+      Fmt.pf fmt "Ebinop binop (%a) (%a)" pp_expr e1 pp_expr e2
+  | Eload (_memory_chunk, e) -> Fmt.pf fmt "Eload mem_chunk (%a)" pp_expr e
+
+let rec pp_stmt fmt stmt =
+  match stmt with
+  | Sskip -> Fmt.pf fmt "[stmt] Sskip"
+  | Sset (id, e) ->
+      Fmt.pf fmt "[stmt] Sset (%d, %a)" (num_to_integer id) pp_expr e
+  | Sstore (chunk, e1, e2) ->
+      Fmt.pf fmt "[stmt] Sstore (%a, %a, %a)" Chunk.pp (Chunk.of_compcert chunk)
+        pp_expr e1 pp_expr e2
+  | Scall _ -> Fmt.pf fmt "[stmt] Scall"
+  | Sbuiltin _ -> Fmt.pf fmt "[stmt] Sbuiltin"
+  | Sseq (s1, s2) -> Fmt.pf fmt "[stmt] Sseq (%a, %a)" pp_stmt s1 pp_stmt s2
+  | Sifthenelse _ -> Fmt.pf fmt "[stmt] Sifthenelse "
+  | Sloop s -> Fmt.pf fmt "[stmt] Sloop (%a)" pp_stmt s
+  | Sblock s -> Fmt.pf fmt "[stmt] Sblock (%a)" pp_stmt s
+  | Sexit _ -> Fmt.pf fmt "[stmt] Sexit "
+  | Sswitch _ -> Fmt.pf fmt "[stmt] Sswitch "
+  | Sreturn _ -> Fmt.pf fmt "[stmt] Sreturn "
+  | Slabel (label, s) ->
+      Fmt.pf fmt "[stmt] Slabel (%d, %a)" (num_to_integer label) pp_stmt s
```

Comment:
Do we still need the printers?
Also, your `num_to_integer` is bound to exist in the `Camlcoq` library

Reply:
Forgot to remove them after debugging

Reply:
No no, Camlcoq is extracted from the coq development of compcert
In particular here, [`label`](https://compcert.org/doc/html/compcert.cfrontend.Csharpminor.html#label) is defined as [`ident`](https://compcert.org/doc/html/compcert.common.AST.html#ident) which is defined as [`positive`](https://compcert.org/doc/html/compcert.common.AST.html#ident) (the coq development is clickable)

A quick look at `Camlcoq` shows that the `P` module, for `positive` values contains a `to_int` function (which is used is [`trans_label`](https://github.com/GillianPlatform/Gillian/blob/master/Gillian-C/lib/gilgen.ml#L191) for example.

So you can replace all calls of `num_to_integer` by `Camlcoq.P.to_int`